### DISCUSSION TITLE
generating colors as an object of functions

### DISCUSSION THREAD

**Discussion by kouji-dev** [tailwindlabs/tailwindcss#5605] [Help]
I have a file called colors.js which has the following

> 
![image](https://user-images.githubusercontent.com/22368199/134810805-24ea6dd1-d46a-4bf5-b24e-cb61e65f06c8.png)

this is the generated file with postcsss command: npm run build

> 
![image](https://user-images.githubusercontent.com/22368199/134810826-8ace8412-7930-4dab-bc77-2c083c5b9a81.png)

It has js inside it, but when using a different config like this:

> 
![image](https://user-images.githubusercontent.com/22368199/134810871-a4144d5c-ade4-4d06-a7c9-41d67d1061ac.png)

it works perfectly and the generated CSS file doesn't have any js inside

**What version of Tailwind CSS are you using?**

tailwindcss: 2.2.15
postcss: 8.3.8
postcss-import: 14.0.2
postcss-nested: 5.0.6
postcss-cli: 9.0.0
autoprefixer: 10.3.5
cssnano: 5.0.8

node: v16.6.0

operating system: Windows

example: https://github.com/goodbeysr/colors-config

**Comment by adamwathan** [comment]
Hey! I think there is probably some subtle bug in the way you have this set up that isn't quite assigning things the values you want. If you can provide an actual minimal reproduction (a GitHub repo with just the bare minimum code to demonstrate this not working) it'll be easier for someone to help point out the change that needs to be made.

**Reply by kouji-dev** [reply]
I updated the git repo link to the bare minimum! Hope someone can help :D

**Reply by kouji-dev** [reply]
@adamwathan  any chance? this is really weird, I followed exactly what's in the docs

**Reply by crswll** [reply]
Not sure if this helps at all but it's working here: https://play.tailwindcss.com/be4arOmJvP?file=config

**Reply by kouji-dev** [reply]
@crswll it is working live, but did you check my repo ? i would love to generate css file for that tailwindcss config you did, therefore im using an npm run build command, and the output is wrong

**Reply by crswll** [reply]
The issue seems to be a bug with defining `corePlugins`.

https://play.tailwindcss.com/gABJC0VwII?file=config

If you uncomment the `corePlugins` line in the play above it stops working.

**Reply by crswll** [reply]
Ok, it's not a bug because you **can** do `corePlugins: ['textColor', 'textOpacity']` and it'll work just fine. You'll just need to add all the right `*Opacity` plugins to the `corePlugins` list.

**Comment by crswll** [comment]
The issue here is that you need to also list the plugins that allows the plugin to have opacity.

You would need to do:
```js
corePlugins: [
  'textColor',
  'textOpacity',
  'backgroundColor',
  'backgroundOpacity',
  ...
],
```

The list at the bottom here should be really helpful: https://tailwindcss.com/docs/configuration#core-plugins
